src/main/java/org/apache/ibatis/executor/resultset/DefaultResultSetHandler.java

```diff
@@ -477,7 +477,7 @@ private Object getPropertyMappingValue(ResultSet rs, MetaObject metaResultObject
           }
         }
         final String property = metaObject.findProperty(propertyName, configuration.isMapUnderscoreToCamelCase());
-        if (property != null && metaObject.hasSetter(property)) {
+        if (property != null && metaObject.hasSetter(property) && !resultMap.getMappedProperties().contains(property)) {
```

Comment:
> Hence i think phone is not unknown column.

Okay, I have no strong opinion about this.

> I think we should consider to apply a mechanism that notify duplicate mapping for same property (like same as autoMappingUnknownColumnBehavior).

The duplicate mapping situation like this would be rare.
And this unknown column detection can be inaccurate [1] anyway, so let's consider when someone desperately wants it ;)

[1] There will be false-positives with a nested result map.